### ISSUE TITLE
[BUG] classNames are ignored in v5.13.0

### ISSUE THREAD

**Issue by ansgarsteinkamp** [ReactTooltip/react-tooltip#1025]
Bug in v5.13.0: classNames like the Tailwind-Class `bg-blue-900` are ignored somehow:

```
<Tooltip className="z-50 bg-blue-900" id="Home" delayShow={200}>
   Home
</Tooltip>
```

With v5.12.0 everything works just fine.

**Comment by danielbarion** [comment]
Hi, I tested here and the issue is:

* CSS is being injected into the page

CSS works like objects, when we pass a custom class, this is the behavior:

Before `v5.13.0`

1. import CSS styles:
2. component receive a new class
3. browser process and render it

```
.tooltip-default-css {
  background: red;
}

.bg-green-500 {
  background: green;
}

--------- same as:

Object.assign({ background: 'red' }, { background: 'green' })
```

On version `v5.13.0`:

1. component receives a new class
2. CSS is injected into the page
4. browser process and render it

```
.bg-green-500 {
  background: green;
}

.tooltip-default-css {
  background: red;
}
--------- same as:

Object.assign({ background: 'green' }, { background: 'red' })
```


this is a `bug`, we will take a look as soon as possible. We are open to suggestions on fixing this because the CSS class from props and injected CSS have the same CSS specificity.

Solution 1 -> you need to increase the CSS specificity from your side - I don't like to force the community using our library to write extra code like this
Solution 2 -> we need to increase the specificity of the CSS received by props (how?) - I prefer this approach, but I don't have any ideas on how to do that at the momentn


reference: https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity

**Comment by gabrieljablonski** [comment]
#1027 suggests a workaround

**Comment by danielbarion** [comment]
yeah, that workaround is just the dev increasing the specificity of his CSS query, it's the solution 1 that I mentioned.

another way to do that together with taiwlind is:

```
HTML - JSX
<div className="container">
  <Tooltip className="tooltip" />
</div>

CSS
.container .tooltip {
  @apply bg-green-500;
}
```

**Comment by Tomekmularczyk** [comment]
Hey guys, I don't think that's problem with the current version. Tailwind classes were ignored with react-tooltip@5.8.3. This is why I had to create separate CSS class like:

```css
.tooltipWrapper { 
  @apply rounded-lgplus bg-tixy-1000 p-2 text-sm font-normal leading-4 text-tixy-10 shadow-sm;
}
```

or with the current version:
```tsx
.tooltipWrapper[role="tooltip"] { 
  @apply rounded-lgplus bg-tixy-1000 p-2 text-sm font-normal leading-4 text-tixy-10 shadow-sm;
}
```

**Comment by ansgarsteinkamp** [comment]
> Tomekmularczyk

There is definitely a problem with the current version 5.13.0. Everything works fine with version 5.12.0.

**Comment by NemeZZiZZ** [comment]
@danielbarion , how to completely remove default styling of tooltip? We use this lib hardly in our project, but every time I need to find some crazy way to override it's internal stylings :( Why just not add `removeStyle` prop for those who can deal with it low-level?

**Comment by gabrieljablonski** [comment]
@NemeZZiZZ There's no equivalent to a `removeStyles` prop, as it would definitely be a rarely used feature. You can just use the `styles` prop to override whatever you need.

```tsx
<Tooltip styles={myCustomStyles} />
```

**Comment by NemeZZiZZ** [comment]
Daniel did it on my request for v4 just before v5.
I think it's not so rare, you just have to create some crazy stuff to avoid style overlappings if styles are totally separated of code in your project and looks completely different.
`style` doesn't make you happy because you have no access to entire tooltip's markup and states this way.
I said this last year, will write it again:`left` and `top` calculations are more than enough to make tooltip work if you have completely different stylings

**Comment by gabrieljablonski** [comment]
@NemeZZiZZ 

That's totally understandable, but by our experience this hasn't been an issue to the package users.

You're welcome to open a PR with a rough idea of how you suggest we do this (doesn't have to be perfect, just enough to get the idea) so we can better discuss it.

**Comment by danielbarion** [comment]
hey @NemeZZiZZ @gabrieljablonski,

I need help here: https://github.com/ReactTooltip/react-tooltip/pull/1041

:)

**Comment by danielbarion** [comment]
@NemeZZiZZ please check: https://github.com/ReactTooltip/react-tooltip/pull/1041#issuecomment-1605726779

**Comment by NemeZZiZZ** [comment]
> @NemeZZiZZ please check: [#1041 (comment)](https://github.com/ReactTooltip/react-tooltip/pull/1041#issuecomment-1605726779)

I'm sorry for being late with no support.
Well, I want to tell, that this just works. Maybe it's not so obvious in daily usage, but styleless and className-based re-styling fully can be achieved at last! That's great, thanks! Surely, it needs to be documented for others - but for now I just wait for next release with no patience :)
Thank you so much, Daniel!

**Comment by danielbarion** [comment]
Available at [v5.16.0](https://github.com/ReactTooltip/react-tooltip/releases/tag/v5.16.0)

I'll close this issue as resolved, @NemeZZiZZ and @ansgarsteinkamp  please feel free to open a new issue if needed,

**Comment by ansgarsteinkamp** [comment]
In version 5.16.1, the issue remains unchanged. `bg-blue-900` is still ignored.
